WordPress/Classes/ViewRelated/Post/PostAuthorSelectorViewController.swift

```diff
@@ -0,0 +1,82 @@
+import UIKit
+
+@objc class PostAuthorSelectorViewController: SettingsSelectionViewController {
+    /// The post to change the author.
+    private var post: AbstractPost!
+
+    /// A completion block that is called after the user selects an option.
+    @objc var completion: (() -> Void)?
+
+    /// Representation of an Author used by the view.
+    private typealias Author = (displayName: String, userID: NSNumber, avatarURL: String?)
+
+    // MARK: - Constructors
+
+    @objc init(_ post: AbstractPost) {
+        self.post = post
+
+        let authors = PostAuthorSelectorViewController.sortedActiveAuthors(for: post.blog)
+
+        guard !authors.isEmpty, let currentAuthorID = post.authorID else {
+            super.init(style: .plain)
+            return
+        }
+
+        let authorsDict: [AnyHashable: Any] = [
+            "DefaultValue": currentAuthorID,
+            "Title": NSLocalizedString("Author", comment: "Author label."),
+            "Titles": authors.map { $0.displayName },
+            "Values": authors.map { $0.userID },
+            "CurrentValue": currentAuthorID
+        ]
+
+        super.init(dictionary: authorsDict)
+
+        onItemSelected = { [weak self] authorID in
+            guard
+                let authorID = authorID as? NSNumber,
+                let author = authors.first(where: { $0.userID == authorID }),
+                !post.isFault, post.managedObjectContext != nil
+            else {
+                return
+            }
+
+            post.authorID = author.userID
+            post.author = author.displayName
+            post.authorAvatarURL = author.avatarURL
+
+            self?.completion?()
+        }
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override init!(style: UITableView.Style, andDictionary dictionary: [AnyHashable: Any]!) {
+        super.init(style: style, andDictionary: dictionary)
+    }
+
+    override init(style: UITableView.Style) {
+        super.init(style: style)
+    }
```

Comment:
Are these overrides needed?

Reply:
My Swift -> ObjC -> Swift is weak here, but we're subclassing `SettingsSelectionViewController` and on `init` of our subclass calling `super.init(dictionary: NSDictionary)`. That method inside of `SettingsSelectionViewController` expects to return an `instancetype` of the return of the call to:

```initWithStyle:(UITableViewStyle)style
                andDictionary:(NSDictionary *)dictionary;```
                
It appears if that's not available in the subclass, we crash. This is also the reason why the `post` property of `PostAuthorSelectorViewController` needs to be an implicitly unwrapped var, otherwise its initialization would be required in the `super.init` call to these methods.

I'm open to ideas - this was mostly boilerplate I took from `PostVisibilitySelectorViewController`.

Reply:
I believe [this is our answer](https://docs.swift.org/swift-book/LanguageGuide/Initialization.html#ID221).

Since we have inits calling other inits via subclass, Swift expects the more "specialized" subclass to have the appropriate initializers.

Reply:
Thanks for the link! That looks like it.

Reply:
Resolving this comment thread now – I think your spot on with the reason why these overrides are needed.